base/gmp.jl

```diff
@@ -699,8 +699,9 @@ function string(n::BigInt; base::Integer = 10, pad::Integer = 1)
     iszero(n) && pad < 1 && return ""
     nd1 = ndigits(n, base=base)
     nd  = max(nd1, pad)
-    sv  = Base.StringVector(nd + isneg(n))
+    sv  = Base.StringVector(nd + isneg(n) + 1) # +1 for final '\0'
     GC.@preserve sv MPZ.get_str!(pointer(sv) + nd - nd1, base, n)
+    @assert iszero(pop!(sv)) # '\0' terminator
```

Comment:
I think we should treat `@assert` as only optionally (like, in a debug build) evaluating its argument, so the `pop!` should probably be separate.